lisa/operating_system.py

```diff
@@ -1520,9 +1527,13 @@ def add_repository(
 
     def _initialize_package_installation(self) -> None:
         self.wait_running_process("zypper")
-        self._node.execute(
+        output = self._node.execute(
             "zypper --non-interactive --gpg-auto-import-keys refresh", sudo=True
-        )
+        ).stdout
+        if self._no_repo_defined.search(output):
+            raise LisaException(
+                f"There are no enabled repositories defined in {self._node.os.name}"
```

Comment:
Is the os name enough to troubleshoot? Is it need the distro version?

Reply:
I'll check.
BTW, two possible reasons here
- one is the disto is byos one, no repo is by design
- the other issue is that register repo has issue

Reply:
add distro version, then it can provide more info